Travel Scheduler/Test/Commute.m

```diff
@@ -0,0 +1,51 @@
+//
+//  Commute.m
+//  Travel Scheduler
+//
+//  Created by gilemos on 7/20/19.
+//  Copyright © 2019 aliu18. All rights reserved.
+//
+
+#import "Commute.h"
+#import "APIManager.h"
+#import "Step.h"
+
+@implementation Commute
+
+-(instancetype)initWithOrigin:(NSString *)originId toDestination:(NSString *)destinationId withDepartureTime:(int)departureTime{
+    self = [self init];
+    self.departureTimeInSecondsSince1970 = departureTime;
+    [[APIManager shared]getCommuteDetailsFromOrigin:originId toDestination:destinationId withDepartureTime:departureTime withCompletion:^(NSArray *commuteInfoArray, NSError *error) {
+        if(commuteInfoArray) {
+            [self initAllProperties];
+            [self buildCommuteWithDictionary:commuteInfoArray[0]];
+            [self makeArrayOfStepsWithRootDictionary:commuteInfoArray[0]];
+        }
+        else {
+            NSLog(@"did not work snif");
```

Comment:
So in final versions of code, you shouldn't have NSLogs. Instead, consider how you can have either fallbacks or failure handling in the most graceful way possible. Depending on the scenario, it can be a default value, re-trying, informing the user and requesting again or even sending back to home screen.
These decisions are what drives user experience and edge case designs for best product quality, and worth having a group discussion about!

Reply:
Yeah. I was really confused on what we should put at those locations when we plan on removing the NSLogs. Thanks for the clarification. We would do well to discuss it before making final version of code.